pkg/manager/nats.go

```diff
@@ -0,0 +1,96 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-project/nats-manager/pkg/k8s"
+	"github.com/kyma-project/nats-manager/pkg/k8s/chart"
+	"go.uber.org/zap"
+)
+
+type NatsConfig struct {
+	ClusterSize int
+}
+
+// Perform a compile time check.
+var _ Manager = &NatsManager{}
+
+//go:generate mockery --name=Manager --outpkg=mocks --case=underscore
+type Manager interface {
+	GenerateNATSResources(*chart.ReleaseInstance, ...Option) (*chart.ManifestResources, error)
+	DeployInstance(context.Context, *chart.ReleaseInstance) error
+	DeleteInstance(context.Context, *chart.ReleaseInstance) error
+	IsNatsStatefulSetReady(context.Context, *chart.ReleaseInstance) (bool, error)
```

Comment:
`GenerateNATSResources` and `IsNatsStatefulSetReady` please either use `Nats` or `NATS`.